src/tiledb/cloud/asset.py

```diff
@@ -1,19 +1,109 @@
 """An asset may be an array or a group."""
 
 from functools import partial
-from typing import Callable, List, Mapping, Optional, Union
+from typing import Callable, List, Literal, Mapping, Optional, Union
 
 import tiledb
 
-from . import array  # type: ignore
-from . import groups  # type: ignore
-from .rest_api.models import ArrayInfo  # type: ignore
-from .rest_api.models import GroupInfo  # type: ignore
+from . import array
+from . import client
+from . import groups
+from .rest_api import models
+from .rest_api.api import assets_api
+
+_AssetType = Union[Literal["array"], Literal["group"]]
+_Depth = Union[Literal["root"], Literal["all"]]
+_OwnershipLevel = Union[Literal["owned"], Literal["shared"]]
+
+
+def list(
+    *,
+    namespace: Optional[str] = None,
+    search: Optional[str] = None,
+    type: Optional[_AssetType] = None,
+    ownership_level: Optional[_OwnershipLevel] = None,
+    depth: Optional[_Depth] = None,
+    expand: Optional[str] = None,
```

Comment:
Should this be a union of a list or string and we handle the conversion to the csv parameter?